### PR TITLE
Maintain formatting and check for lint issues on CI

### PR DESCRIPTION
**Pull request by s0undt3ch**
* Solves a few lint issues
* disables a few other lint checks(don't want to change project's naming convention for example)
* Black the code
* Enforce all of the above for each PR

### CODE REVIEW

**Comment by j-waters** [comment]
I definitely think enforcing a code style is a good idea, and this PR is a lot more thorough than my old one #95 so I'll close it

**Comment by segersniels** [comment]
Would suggest to add the linting to the already present workflow so you can make the actual publishing of the docker image depend on the linting (splitting it up into two jobs).

File: .github/workflows/testing.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+name: CI
```
Comment (segersniels) [inline .github/workflows/testing.yml]: Would suggest to add the linting to the already present workflow so you can make the actual publishing of the docker image depend on the linting (splitting it up into two jobs).

**Review by segersniels — APPROVED** [review]

File: .github/workflows/testing.yml (hunk 1 above)
Comment (s0undt3ch) [reply .github/workflows/testing.yml]: Then linting would only run on release?

Or you only actually prefer the CI and Release jobs to be on the same workflow?

Comment (s0undt3ch) [reply .github/workflows/testing.yml]: Nevermind, will do.

Comment (s0undt3ch) [reply .github/workflows/testing.yml]: Done

**Comment by s0undt3ch** [comment]
How about even building the containers, to confirm it works, but only push the containers on "push" events?

**Comment by s0undt3ch** [comment]
@edeng23 is this PR welcomed?
Rebasing for every change going in is not that cool, so, if not desirable, I'll just close it. I'll keep updating/rebasing otherwise.

File: .github/workflows/testing.yml (hunk 1 above)
Comment (segersniels) [reply .github/workflows/testing.yml]: Would just add a conditional to the deploy job to only run on master branch while allowing linting to run against everything.

Eg.

`if: github.ref == 'refs/heads/master'`

Comment (segersniels) [reply .github/workflows/testing.yml]: Also just keep the original workflow file. No need to add a rename into this PR when it's already there.

Comment (s0undt3ch) [reply .github/workflows/testing.yml]: It's no longer release only focused....

Comment (s0undt3ch) [reply .github/workflows/testing.yml]: Are you sure?

Comment (s0undt3ch) [reply .github/workflows/testing.yml]: > Would just add a conditional to the deploy job to only run on master branch while allowing linting to run against everything.

I had it like that at one point.
But then you don't test building the containers....

**Comment by edeng23** [comment]
@s0undt3ch hey, yes, your PR is welcomed. I'll take a close look at the linting rules and merge as soon as I can. Thanks!

File: .github/workflows/testing.yml (hunk 1 above)
Comment (segersniels) [reply .github/workflows/testing.yml]: That's how it's usually done. Usually want the action to come back as positive as fast as possible without having to wait on a multi arch image to build.

Comment (s0undt3ch) [reply .github/workflows/testing.yml]: I agree, as long as there's a "state" where we test if we can create the containers before releasing.

Ie, I'll be onboard with that if we push the docker containers as "dev" versions(or not publish at all) and we start versioning and tagging proper releases in git. Probably a `stable` tag on these release containers would also be suitable.

**Comment by edeng23** [comment]
@s0undt3ch I'm ready to merge this PR. The only code style linting feature that I don't like is the separation of listed imports. Other than that everything looks good :)
Can you please update the .pylintrc file and rebase?
Thanks!

**Comment by s0undt3ch** [comment]
That's done by the reorder python imports pre-commit hook, which also removed duplicates and separates app imports from stdlib imports from 3rd-party imports.
That also means redo most of the PR from scratch. Not impossible since some of this work is does by pre-commit and the installed hooks.

Anyway, I've moved on. If I do find time, I'll do it, but of someone wants to take over, please do.

**Comment by j-waters** [comment]
I've created https://github.com/edeng23/binance-trade-bot/pull/202, which implements your suggestion @edeng23, fixes some config issues, adds some documentation, and is fully rebased

**Pull request closed without merging**